### PR TITLE
Remove necessity to re-generate requirements files

### PR DESCRIPTION
**Pull request by Carreau**
## Description

This remove the needs to keep those files in sync with pyproject.toml. 

I think we can also remove them and reference directly `.[target,...]` where relevant, but I don't know if there are external things (feedstock, readthedocs conf) that reference those files, so I haven't touched them. 

At least the ci recipe could use `.[target,...]` already

## Release note

```release-note
...
```

### CODE REVIEW

**Comment by stefanv** [comment]
Great, thanks @Carreau! Let's go ahead and also update the CI calls, docs, and remove the requirements files, if you're up for it. Needs a rebase.

File: requirements/build.txt [hunk 1]
```diff
@@ -1,12 +1 @@
...
-wheel
-setuptools>=67
-packaging>=21
-ninja
-Cython>=3.0.4
-pythran
-numpy>=2.0.0rc1
-spin==0.8
-build
+.[build]
```
Comment (lagru) [inline requirements/build.txt:1]: Maybe I'm missing something, but I don't think this is an equivalent replacement. If I run `pip install -r requirements/build.txt` it installs scikit-image as well where previously we could setup the build environment without having to install scikit-image as well. Not sure if we want or need to keep this ability.

If not, then why not invoke `pip install .[build]` directly and forgo requirement files alltogether?

Comment (stefanv) [reply requirements/build.txt:1]: Err, you are right; this is not what we want.

The alternative approach (where everything is defined in `pyproject.toml`) is [install-deps](https://gist.github.com/stefanv/0b052fa4014fa07e18e81fe544afc9f9).

Comment (Carreau) [reply requirements/build.txt:1]: > If not, then why not invoke `pip install .[build]` directly and forgo requirement files alltogether?

That what I would have done in the next step, but I don't know if external services (readthedocs) have configuration that point to those files, so I didn't wanted to remove them.

**Pull request closed without merging**